api/client/proto/authservice.proto

```diff
@@ -1798,6 +1809,9 @@ service AuthService {
 
     // GetWindowsDesktops returns all registered Windows desktop hosts.
     rpc GetWindowsDesktops(google.protobuf.Empty) returns (GetWindowsDesktopsResponse);
+    // GetWindowsDesktopsByName returns all registered Windows desktop hosts matching name.
+    rpc GetWindowsDesktopsByName(GetWindowsDesktopsByNameRequest)
```

Comment:
Do we need this new rpc?

Can we just use `GetWindowsDesktop()` where we set name and leave `HostID` as empty string?

Reply:
I think its a little clearer/makes calls a little nicer, to have separate rpcs as `GetWindowsDesktop` always returnes a single item and `GetWindowsDesktopsByName` always expects a list.

Do other rpcs follow a similar pattern of "" acting like a glob key or something?

Reply:
It's not so much of a glob as it is a filter.

- `rpc ListNodes(ListNodesRequest)` lists nodes that meet the criteria of the request.
- `rpc GetAccessRequests(AccessRequestFilter)` gets access requests filtered by the paramters in the request, and unset means no filter, just like I proposed here
- `rpc GetPluginData(PluginDataFilter)` behaves similarly.

Reply:
Ah thats pretty cool, maybe it would be worth changing it so instead of having multiple rpc endpoints for getting desktops, there could just be GetWindowsDesktops that has a filter type for requests instead.